### PR TITLE
Fix release script and CI config

### PR DESCRIPTION
**Pull request by juliaogris**
Fixes 58# and 43# (hopefully).

Changes proposed in this pull request:
- Update Appveyor encrypted OAuth token (fingers crossed).
- Improve release script style
- Add automatic deployment to PyPI from Travis on tags.

@anz-bank/sysl-developers

### CODE REVIEW

**Comment by juliaogris** [comment]
Annoyingly something is broken with Appveyor creating Github releases for organisations. 
I've created an [issue](https://github.com/appveyor/ci/issues/2068) on the Appveyor issue tracker and the [forum](https://help.appveyor.com/discussions/problems/11865-deployment-to-github-releases-for-organisation-seems-broken)

File: src/scripts/release.sh [hunk 1]
```diff
@@ -23,20 +23,20 @@ fi
 
 ORIGIN_URL=$(git remote get-url origin)
 UPSTREAM="anz-bank"
-UPSTREAM_URL=$(echo "$ORIGIN_URL" | sed -n "s/\\(.*\\/\\/github.com\\/\\)\\(..*\\)\\(\\/..*.git\\)/\\1$UPSTREAM\\3/p")
+UPSTREAM_URL=$(echo "$ORIGIN_URL" | sed -n "s|\\(.*//github.com/\\)\\(..*\\)\\(/..*.git\\)|\\1$UPSTREAM\\3|p")
```
Comment (camh-anz) [inline src/scripts/release.sh]: My eyes are still bleeding, even with the slashes now non-escaped.

I suggest using single quotes and Extended Regular Expressions (EREs) with sed to make this cleaner (there's a missing escape for the dot-git, which will mean one backslash will survive, so single-quotes will stop this doubling-up):

`sed -E -n 's|(.*//github.com/)(.+)(/.+\.git)|\1'"${UPSTREAM}"'\3|p`

Maybe I'm missing something, but if $ORIGIN_URL is a single URL, you don't need `sed -n` with the `p` modifier.

File: src/scripts/release.sh [hunk 2]
```diff
@@ -23,20 +23,20 @@ fi
...
 echo "------- Pull upstream ---------"
 git pull "$UPSTREAM_URL" master || fatal "Cannot pull  upstream master"
 
-if [[ $COMANND = "prepare" ]]; then
+if [[ $COMMAND = "prepare" ]]; then
     RELEASE_BRANCH="release-v$VERSION"
-    ORIGIN=$(echo "$ORIGIN_URL" | sed -n "s/.*\\/\\/github.com\\/\\(..*\\)\\/..*.git/\\1/p")
-    REPO=$(echo "$ORIGIN_URL" | sed -n "s/.*\\/\\/github.com\\/..*\\/\\(..*\\).git/\\1/p")
+    ORIGIN=$(echo "$ORIGIN_URL" | sed -n "s|.*//github.com/\\(..*\\)/..*.git|\\1|p")
+    REPO=$(echo "$ORIGIN_URL" | sed -n "s|.*//github.com/..*/\\(..*\\).git|\\1|p")
```
Comment (camh-anz) [inline src/scripts/release.sh]: Some comment here re EREs, single quotes and `sed -n`

File: src/scripts/release.sh (hunk 1 above)
Comment (juliaogris) [reply src/scripts/release.sh]: Much nicer, thanks. Done. 
(I can't possibly be held responsible for bleeding eyes).

File: src/scripts/release.sh (hunk 2 above)
Comment (juliaogris) [reply src/scripts/release.sh]: Done.

File: src/scripts/release.sh [hunk 3]
```diff
@@ -23,20 +23,20 @@ fi
 
 ORIGIN_URL=$(git remote get-url origin)
 UPSTREAM="anz-bank"
-UPSTREAM_URL=$(echo "$ORIGIN_URL" | sed -n "s/\\(.*\\/\\/github.com\\/\\)\\(..*\\)\\(\\/..*.git\\)/\\1$UPSTREAM\\3/p")
+UPSTREAM_URL=$(echo "$ORIGIN_URL" | sed -E 's|(.*//github.com/)(.*)(/.*\.git)|\1'"${UPSTREAM}"'\3|')
```
Comment (camh-anz) [inline src/scripts/release.sh]: You may as well make this $UPSTREAM and not ${UPSTREAM} as the rest of the script uses the first form. I always use the second and I type it without thinking which is why I said it last time - it makes no difference most of the time.

Also, did `+` not work in the regex? I recall having issues with it in the past, but if you're not using `.+`, you should use `..*` as you were before.

Comment (juliaogris) [reply src/scripts/release.sh]: Argh - sorry, just sloppy on my side. Something else didn't work blablabla
Done.

File: src/scripts/release.sh [hunk 4]
```diff
@@ -23,20 +23,20 @@ fi
 
 ORIGIN_URL=$(git remote get-url origin)
 UPSTREAM="anz-bank"
-UPSTREAM_URL=$(echo "$ORIGIN_URL" | sed -n "s/\\(.*\\/\\/github.com\\/\\)\\(..*\\)\\(\\/..*.git\\)/\\1$UPSTREAM\\3/p")
+UPSTREAM_URL=$(echo "$ORIGIN_URL" | sed -E 's|(.*//github.com/)(.+)(/.+\.git)|\1'"$UPSTREAM"'\3|')
```
Comment (marcelocantos) [inline src/scripts/release.sh]: Needs an explanatory comment

File: src/scripts/release.sh [hunk 5]
```diff
@@ -23,20 +23,20 @@ fi
...
 
 echo "------- Pull upstream ---------"
 git pull "$UPSTREAM_URL" master || fatal "Cannot pull  upstream master"
 
-if [[ $COMANND = "prepare" ]]; then
+if [[ $COMMAND = "prepare" ]]; then
     RELEASE_BRANCH="release-v$VERSION"
-    ORIGIN=$(echo "$ORIGIN_URL" | sed -n "s/.*\\/\\/github.com\\/\\(..*\\)\\/..*.git/\\1/p")
-    REPO=$(echo "$ORIGIN_URL" | sed -n "s/.*\\/\\/github.com\\/..*\\/\\(..*\\).git/\\1/p")
+    ORIGIN=$(echo "$ORIGIN_URL" | sed -E 's|.*//github.com/(.+)/.+\.git|\1|')
```
Comment (marcelocantos) [inline src/scripts/release.sh]: Ditto

File: src/scripts/release.sh (hunk 4 above)
Comment (juliaogris) [reply src/scripts/release.sh]: ? This is a merged PR, I've implemented the changes Cam suggested on chat. Do you want me to write that here? Functionality doesn't change, just less bleeding eyes for @camh-anz as he put it.

**Pull request merged**